examples/client_objects.rs

```diff
@@ -0,0 +1,51 @@
+use azure_identity::{
+    ManagedIdentityCredential,
+    ManagedIdentityCredentialOptions,
+    UserAssignedId
+};
+use azure_security_keyvault_secrets::{
+    SecretClient, 
+    SecretClientOptions
+};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+
+    let user_assigned_id_value = std::env::var("AZURE_USER_ASSIGNED_IDENTITY")
+        .map_err(|_| {
+            std::io::Error::new(
+                std::io::ErrorKind::NotFound,
+                "AZURE_USER_ASSIGNED_IDENTITY environment variable is required",
+            )
+        })?;
+
+    let user_assigned_id = UserAssignedId::ClientId(user_assigned_id_value);
+
+    let credential_options = ManagedIdentityCredentialOptions {
+        user_assigned_id: Some(user_assigned_id),
+        ..Default::default()
+    };
+
+    let credential = ManagedIdentityCredential::new(Some(credential_options))?;
+
+    let key_vault_endpoint = std::env::var("AZURE_KEY_VAULT_ENDPOINT")
+        .map_err(|_| {
+            std::io::Error::new(
+                std::io::ErrorKind::NotFound,
+                "AZURE_KEY_VAULT_ENDPOINT environment variable is required",
+            )
+        })?;
```

Comment:
This is more complicated than necessary. Since `&'static str` implements `std::error::Error`, I'd just emit a string like you've done elsewhere and don't wrap it in an error you don't own anyway.